src/cfnlint/rules/resources/route53/RecordSet.py

```diff
@@ -32,184 +32,163 @@ class RecordSet(CloudFormationLintRule):
     REGEX_TXT = re.compile(r'^("[^"]{1,255}" *)*"[^"]{1,255}"$')
     REGEX_CNAME_VALIDATIONS = re.compile(r'^.*\.acm-validations\.aws\.?$')
 
-    def check_a_record(self, path, recordset):
-        """Check A record Configuration"""
+    def count_c_names(self, records, path, cfn):
+        """ Count C Names """
         matches = []
 
-        resource_records = recordset.get('ResourceRecords')
-        for index, record in enumerate(resource_records):
+        scenarios = cfn.get_object_without_nested_conditions(records, path)
+        for scenario in scenarios:
+            if len(scenario.get('Object')) > 1:
+                scenario = scenario.get('Scenario')
+                message = 'A CNAME recordset can only contain 1 value'
+                if scenario is None:
+                    message = 'A CNAME recordset can only contain 1 value'
```

Comment:
Shouldn't that message be something like "A CName recordset must contain 1 value"?